docs/api_proposal.md

```diff
@@ -0,0 +1,146 @@
+# Taxonomy API proposal
+
+Design document for DINA taxonomy API methods. Expanded from [Taxonomy API discussion](https://docs.google.com/spreadsheets/d/1uaJh1qt6mvY0ZCEB6uBKdnUnyGdpQnZcnFqTMHTS5pA/edit#gid=0) google doc.
+
+In this doc:
+* [Assumptions](#assumptions)
+* [Methods](#methods)
+* [Questions](#questions)
+
+See [DINA glossary](https://github.com/DINA-Web/dina-use-cases/blob/master/glossary.md) for terms.
+
+See [API example doc](https://github.com/DINA-Web/taxonomy/blob/master/docs/api_examples.md) for notes on non-DINA taxonomy APIs
+
+## Assumptions
+
+* taxonomy = hierarchy = classification = checklist
+* there will be some canonical root URL for the API, format to be determined
+* the API will be versioned, and we may deprecate older versions in the future
+* exact format / fields of result TBD; examples below are just for discussion about general form of result for each method
+* taxonomic names are not guaranteed to be unique
+  * therefore, cannot safely perform CRUD options based on name strings without disambiguation
+* one goal is to harmonize with existing taxonomy APIs (improved interoperability with other services and future modules)
+
+## Methods
+
+Starting with GET methods. At a high level, we need methods to
+1. Search for taxa based on name strings (taxonomic name resolution)
+1. Search taxa based on some parameter
+1. Get a specific  taxon
+1. Return information about the taxonomic hierarchy / relationships between taxa.
+
+Here is a proposed list of key GET methods:
+
+### /tnrs
+Method for taxonomic name reconciliation; returns potentially matching taxa (exact or fuzzy matching; uses information about synonyms)
```

Comment:
It could be interesting (at least I think) to link back to the use case e.g. https://github.com/DINA-Web/dina-use-cases/blob/master/taxonomy/16-Search_taxon_names.md

Reply:
To be the flavour of RESTful web services we are using, end poiints should not be verbs, but nouns. Verbs are filters. As per http://jsonapi.org/recommendations/#filtering
In this case, taking the noun to be taxon, you could have:
/taxon?filter[tnrs]=....